### DISCUSSION TITLE
Private Gists Not Returning

### DISCUSSION THREAD

**Discussion by jobvite-github** [octokit/octokit.js#1867] [Q&A] [answered]
I have my authenticated API call to `/gists` working, but I don't get any **_private_** gists back, only the authenticated user's **_public_** gists. I don't know if this is a bug, or if I'm missing a piece of the puzzle. I have also tried using Postman, with a header `Authentication: <token>`, and still only got the authenticated user's public gists back from the API. So, maybe it's a GitHub API thing? 

Any insight would be greatly appreciated. I have been teaching myself how to incorporate Octokit into a web app, and I wouldn't be overly surprised if I'm missing something; I just don't know what that might be at this point.

**Comment by gr2m** [comment, marked as answer]
Can you share the error your are getting with all properties (you will have to `console.log(error)`). Most likely you didn't enable the `gist` scope for the personal access token

**Reply by jobvite-github** [reply]
I'm not getting an error, as far as I know.

**Reply by jobvite-github** [reply]
I should also add, that I do have the `gist` scope enable for my personal access token.

**Reply by gr2m** [reply]
Sorry, I thought you might have tried to request a private gist directly, I assume you'd get a 404 in that case. 

Can you please try to create a new token with the `gist` scope enabled and use that?
https://github.com/settings/tokens/new?scopes=gist

**Reply by jobvite-github** [reply]
I just created another token with the gist scope enable, but the same outcome. 😞

**Reply by jobvite-github** [reply]
Wait... The token I'm using for authenticated API calls is the one I get returned from GitHub. Is that not correct?

Copying the token value created by GitHub in Postman returned the right response.

**Reply by gr2m** [reply]
> The token I'm using for authenticated API calls is the one I get returned from GitHub. Is that not correct?

what do you mean with "returned by GitHub"?

**Reply by jobvite-github** [reply]
I authenticate the user through a GitHub OAuth App, which redirects the user to GitHub to login, which redirects them back to my app with an authentication token for the user.

https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/#web-application-flow

**Reply by jobvite-github** [reply]
I guess a better way to ask my question, now realizing an `OAuth Token`'s scope is different (_somehow_) than the `Personal Access Token`:

Can I use Octokit with an OAuth Token and still get the access I need to `secret` gists?

**Reply by gr2m** [reply]
In that case the OAuth app has to request the gist scope. Are you in control of the OAuth app?

**Reply by jobvite-github** [reply]
Yes, I own the OAuth app.

**Reply by jobvite-github** [reply]
@gr2m Thank you thank you thank you! That little bit of info what exactly what I needed. All I needed to do was add a url parameter, "`scope=gist`" to get something like: `https://github.com/login/oauth/authorize?client_id=<my_client_id>&scope=gist` and I had the correct scope for the API call!

**Reply by gr2m** [reply]
Glad you got it working! We have some OAuth libraries that might help:

https://github.com/octokit?q=OAuth&type=&language=

**Comment by gr2m** [comment]
Here is a test case, I am not able to reproduce the problem:
https://runkit.com/gr2m/octokit-rest-js-1867/1.0.0
